File: back/node/app/delivery/middlewares/Cors.ts
import {Request, Response, NextFunction} from 'express'
const corsMiddleware = (req: Request, res: Response, next: NextFunction) => {

    const corsOrigin = process.env.CORS_ORIGIN || 'http://localhost:8085';

    res.header('Access-Control-Allow-Origin', corsOrigin);
    res.header('Access-Control-Allow-Methods', 'GET,HEAD,OPTIONS,POST,PUT');
    res.header('Access-Control-Allow-Headers', '*');
    res.header('Access-Control-Allow-Credentials', 'true');

    if (req.method === 'OPTIONS') {
        console.log('this is a preflight');
        res.sendStatus(200);
    } else {
        next();
    }
};

export {corsMiddleware}

File: back/go/app/delivery/middlewares/Cors.go
package middlewares

import (
	"fmt"
	"net/http"
	"os"
)

func CorsMiddleware() func(http.Handler) http.Handler {
	return func(next http.Handler) http.Handler {
		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {

			fmt.Println("Setting CORS y")

			corsOrigin := os.Getenv("CORS_ORIGIN")
			if corsOrigin == "" {
				corsOrigin = "http://localhost:8085"
			}

			w.Header().Set("Access-Control-Allow-Origin", corsOrigin)
			w.Header().Set("Access-Control-Allow-Methods", "POST, GET, OPTIONS")
			w.Header().Set("Access-Control-Allow-Headers", "hx-target, hx-current-url, hx-trigger, hx-request, hx-boost, hx-ext, hx-get, hx-swap, Accept, Content-Type, Content-Length, Accept-Encoding, X-CSRF-Token, Authorization")
			w.Header().Set("Access-Control-Allow-Credentials", "true")

			if r.Method == http.MethodOptions {
				println("this is a preflight")
				w.WriteHeader(http.StatusOK)
				return
			}

	
			next.ServeHTTP(w, r)
		})
	}
}